nemo/collections/asr/models/hybrid_rnnt_ctc_bpe_models_tgt_lang.py

```diff
@@ -0,0 +1,1171 @@
+
+        # Setup decoding object
+        self.decoding = RNNTBPEDecoding(
+            decoding_cfg=self.cfg.decoding,
+            decoder=self.decoder,
+            joint=self.joint,
+            tokenizer=self.tokenizer,
+        )
+
+        # Setup wer object
+        self.wer = WER(
+            decoding=self.decoding,
+            batch_dim_index=0,
+            use_cer=self.cfg.get('use_cer', False),
+            log_prediction=self.cfg.get('log_prediction', True),
+            dist_sync_on_step=True,
+        )
+
+        # Setup bleu object
+        self.bleu = BLEU(decoding=self.decoding, tokenize=self.cfg.get('bleu_tokenizer', "13a"), log_prediction=True)
+
+        # Setup fused Joint step if flag is set
+        if self.joint.fuse_loss_wer:
+            self.joint.set_loss(self.loss)
+            self.joint.set_wer(self.wer)
+
+        # Setup CTC decoding
+        ctc_decoding_cfg = self.cfg.aux_ctc.get('decoding', None)
+        if ctc_decoding_cfg is None:
+            ctc_decoding_cfg = OmegaConf.structured(CTCBPEDecodingConfig)
+            with open_dict(self.cfg.aux_ctc):
+                self.cfg.aux_ctc.decoding = ctc_decoding_cfg
+        self.ctc_decoding = CTCBPEDecoding(self.cfg.aux_ctc.decoding, tokenizer=self.tokenizer)
+
+        # Setup CTC WER
+        self.ctc_wer = WER(
+            decoding=self.ctc_decoding,
+            use_cer=self.cfg.aux_ctc.get('use_cer', False),
+            dist_sync_on_step=True,
+            log_prediction=self.cfg.get("log_prediction", False),
+        )
+
+        # setting the RNNT decoder as the default one
+        self.cur_decoder = "rnnt"
+
+        if hasattr(self.cfg, 'initialize_target_lang_id_concatination'):
+            self.concat = True
+            self.num_langs = self._cfg.model_defaults.get('num_langs', 128)
+
+            # Setup normalization
+            self.norm = self._cfg.get('norm', None)
+            if self._cfg.get('norm') == 'ln':
+                self.asr_norm = torch.nn.LayerNorm(self._cfg.model_defaults.enc_hidden)
+                self.lang_norm = torch.nn.LayerNorm(self.num_langs)
+
+            # Setup projection layers
+            proj_in_size = self.num_langs + self._cfg.model_defaults.enc_hidden
+            proj_out_size = self._cfg.model_defaults.enc_hidden
+
+            self.lang_kernal = torch.nn.Sequential(
+                torch.nn.Linear(proj_in_size, proj_out_size * 2),
+                torch.nn.ReLU(),
+                torch.nn.Linear(proj_out_size * 2, proj_out_size),
+            )
+
+    def _setup_dataloader_from_config(self, config: Optional[Dict]):
+
+        if config.get("use_lhotse"):
+            if 'initialize_target_lang_id_concatination' in self.cfg:
+                dataset = LhotseSpeechToTextBpeDatasetTgtLangID(tokenizer=self.tokenizer, cfg=config)
+            else:
+                dataset = LhotseSpeechToTextBpeDataset(tokenizer=self.tokenizer)
+            return get_lhotse_dataloader_from_config(
```

Comment:
This is not a good practice which was followed for previous models. instead of returining here either: Put it in if and else case later part.